### PR TITLE
Activate first restlet1x plugin in order to not get siesta plugin activated before restlet

### PR DESCRIPTION
**Pull request by adreghiciu**
Without this, if siesta plugin activates first, siesta resources are not visible

Signed-off-by: Alin Dreghiciu adreghiciu@gmail.com

### CODE REVIEW

**Comment by nabcos** [comment]
Hm, that's a pretty big hack IMO. +1 as a workaround, but is there no better solution to fix this?

**Comment by adreghiciu** [comment]
Not for the moment. We plan to discuss it with Stuart when he's back, yet as I found out by searching the internet, when using multiple servlet modules with child injectors (as in our case), guice servlet behaviour is "undefined".

**Comment by jdillon** [comment]
Unless Stuart has some magic up his sleeve, we might want to build some sort of bridge between ServletModule and plugins contributing servlets/filters... so that we can have complete control over the behavior.

Ie. single ServletModule in NX core and contributed components to enrich it after plugins have loaded.

File: nexus-core/src/main/java/org/sonatype/nexus/plugins/DefaultNexusPluginManager.java [hunk 1]
```diff
@@ -141,6 +139,21 @@ public DefaultNexusPluginManager( final RepositoryTypeRegistry repositoryTypeReg
         Map<GAVCoordinate, PluginMetadata> filteredPlugins =
             filterInstalledPlugins( repositoryManager.findAvailablePlugins() );
 
+        // TODO HACK!
+        // activate first restlet1x plugin in order to not get siesta plugin activated before restlet
+        // without this, if siesta plugin activates first, siesta resources are not visible
```
Comment (jdillon) [inline nexus-core/src/main/java/org/sonatype/nexus/plugins/DefaultNexusPluginManager.java:144]: You really should also at least hint as to why this happens.  The comment alone makes it appear as if siesta is broken... when in fact the multiple Guice ServletModules are causing unpredictable behavior.

**Pull request merged**